### PR TITLE
API keys: Remove state hideAPIkeys and refactor interface to IsDisabled

### PR DESCRIPTION
**Pull request by eleijonmarck**
**What is this feature?**
- removes the state hideAPIkeys from frontend and backend
- refactors interface to IsDisabled instead of CountAPIkeys
- removes the controller from the APIkeys page

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/serviceaccounts/manager/store.go [hunk 1]
```diff
@@ -7,17 +7,6 @@ import (
 	"github.com/grafana/grafana/pkg/services/serviceaccounts"
 )
 
-/*
```
Comment (eleijonmarck) [inline pkg/services/serviceaccounts/manager/store.go:10]: unnecessary documentation

File: public/app/features/api-keys/MigrateToServiceAccountsCard.tsx [hunk 2]
```diff
@@ -26,27 +27,41 @@ export const MigrateToServiceAccountsCard = ({ onMigrate, disabled }: Props): JS
...
+              confirmText={'Yes, migrate now'}
+              onConfirm={onMigrate}
+              onDismiss={() => setIsModalOpen(false)}
+              confirmVariant="primary"
+              confirmButtonVariant="primary"
+            />
+          </div>
+        </Alert>
+      )}
+      {apikeysCount === 0 && (
```
Comment (eleijonmarck) [inline public/app/features/api-keys/MigrateToServiceAccountsCard.tsx:54]: this is incase the users goes to the apikeys page and we have no apikeys present.

File: pkg/services/navtree/navtreeimpl/admin.go [hunk 3]
```diff
@@ -79,15 +79,11 @@ func (s *ServiceImpl) getOrgAdminNode(c *contextmodel.ReqContext) (*navtree.NavL
 		})
 	}
 
-	hideApiKeys, _, _ := s.kvStore.Get(c.Req.Context(), c.OrgID, "serviceaccounts", "hideApiKeys")
-	apiKeys, err := s.apiKeyService.CountAPIKeys(c.Req.Context(), c.OrgID)
+	disabled, err := s.apiKeyService.IsDisabled(c.Req.Context(), c.OrgID)
```
Comment (eleijonmarck) [inline pkg/services/navtree/navtreeimpl/admin.go:82]: @kalleep here is the new interface abstraction

File: pkg/services/apikey/apikey.go [hunk 4]
```diff
@@ -7,11 +7,12 @@ import (
...
 	GetAPIKeys(ctx context.Context, query *GetApiKeysQuery) error
 	GetAllAPIKeys(ctx context.Context, orgID int64) ([]*APIKey, error)
-	CountAPIKeys(ctx context.Context, orgID int64) (int64, error)
 	DeleteApiKey(ctx context.Context, cmd *DeleteCommand) error
 	AddAPIKey(ctx context.Context, cmd *AddCommand) error
 	GetApiKeyById(ctx context.Context, query *GetByIDQuery) error
 	GetApiKeyByName(ctx context.Context, query *GetByNameQuery) error
 	GetAPIKeyByHash(ctx context.Context, hash string) (*APIKey, error)
 	UpdateAPIKeyLastUsedDate(ctx context.Context, tokenID int64) error
+	// IsDisabled returns true if the API key is available for use.
```
Comment (alexanderzobnin) [inline pkg/services/apikey/apikey.go:16]:
```suggestion
	// IsDisabled returns true if the API key is not available for use.
```

File: public/app/features/api-keys/ApiKeysPage.tsx [hunk 5]
```diff
@@ -128,42 +118,33 @@ export class ApiKeysPageUnconnected extends PureComponent<Props, State> {
...
-                        onMigrate={this.onMigrateApiKey}
-                        onDelete={this.onDeleteApiKey}
-                      />
-                    </VerticalGroup>
-                  ) : null}
-                </>
-              );
-            }}
-          </ApiKeysController>
+          <>
```
Comment (alexanderzobnin) [inline public/app/features/api-keys/ApiKeysPage.tsx:125]: I'm not 100% sure, but this container maybe not needed here (if <Page.Contents /> allow multiple elements inside).

**Review by alexanderzobnin — APPROVED** [review]
Looks good!

File: public/app/features/api-keys/ApiKeysPage.tsx (hunk 5 above)
Comment (eleijonmarck) [reply public/app/features/api-keys/ApiKeysPage.tsx:125]: i see. il try removing it

**Pull request merged**